controllers/clusterversion_controller.go

```diff
@@ -436,3 +445,40 @@ func (c *ClusterVersionReconciler) ensureConsolePlugin() error {
 
 	return nil
 }
+
+func (c *ClusterVersionReconciler) reconcileCSIAddonsSubscription() error {
+
+	csiAddonsSubscription, err := c.getSubscription(fmt.Sprintf("%s.%s", csiAddonsSubscriptionLabelKeyPrefix, c.OperatorNamespace))
+	if err != nil {
+		return fmt.Errorf("failed to get csi addons subscription: %v", err)
+	}
+	ocsClientOperatorSubscription, err := c.getSubscription(fmt.Sprintf("%s.%s", ocsClientOperatorSubscriptionLabelKeyPrefix, c.OperatorNamespace))
+	if err != nil {
+		return fmt.Errorf("failed to get ocs client operaror subscription: %v", err)
+	}
+	if csiAddonsSubscription.Spec.Channel != ocsClientOperatorSubscription.Spec.Channel {
+		csiAddonsSubscription.Spec.Channel = ocsClientOperatorSubscription.Spec.Channel
+		if err = c.Update(c.ctx, csiAddonsSubscription); err != nil {
+			return fmt.Errorf("failed to update csi addons subscription: %v", err)
+		}
+	}
+	return nil
+}
+
+func (c *ClusterVersionReconciler) getSubscription(subscriptionLabelKey string) (*opv1a1.Subscription, error) {
+	subscriptions := &opv1a1.SubscriptionList{}
+	err := c.List(
+		c.ctx,
+		subscriptions,
+		client.InNamespace(c.OperatorNamespace),
+		client.MatchingLabels{subscriptionLabelKey: ""},
+		client.Limit(1),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list subscriptions for %s using labels: %v", subscriptionLabelKey, err)
+	}
+	if len(subscriptions.Items) == 1 {
```

Comment:
instead of label can we go with known name which we use for subscription?

Reply:
The name can be different, But the package name is unique identity under the spec and we use that in the odf.